crates/astria-conductor/src/executor.rs

```diff
@@ -219,7 +241,7 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
             return Ok(Some(execution_hash.clone()));
         }
 
-        let prev_block_hash = self.execution_state.clone();
+        let prev_block_hash = self.commitment_state.firm.clone().unwrap().hash;
```

Comment:
This should be the most recent soft block hash. `execute_block()` is primarily called to execute a block at the tip of the chain, so the most recent block will be the most recent soft block, but may happen to be firm.
In the case that the chain is finalize all the way to the tip, `firm` and `soft` will be the same, but I don't think that is that likely to happen very often.